src/history/StateSnapshot.cpp

```diff
@@ -50,12 +50,14 @@ StateSnapshot::StateSnapshot(Application& app, HistoryArchiveState const& state)
 void
 StateSnapshot::makeLive()
 {
+    auto i = 0;
     for (auto& hb : mLocalState.currentBuckets)
```

Comment:
I think it would be cleaner to use `i` as the index in the loop and access `hb` using a direct lookup `mLocalState.currentBuckets[i]` instead of tracking the same thing in two different ways

Reply:
I recommend using `mLocalState.currentBuckets.at(i)` since `std::vector::at` has bounds checking whereas `std::vector::operator[]` does not.

Reply:
I do not thing we need to check bounds when we do i = 0; i < mLocalState.currentBuckets.size();, so I'll just use []